File: scratchpad/failed-jobs/2026-06-11__17-46-30/ai_agent_mcp_tool_definition__qz75eHA/artifacts/user/wm-agent/f/agent_zr-qz75eha/lookup_sku.ts
// Tool script: lookup_sku
// Returns unit price and currency for a given product SKU

export async function main(sku: string): Promise<{ sku: string; unit_price: number; currency: string }> {
  // Simulated product catalog
  const catalog: Record<string, { unit_price: number; currency: string }> = {
    "WIDGET-42": { unit_price: 9.99, currency: "USD" },
    "GADGET-01": { unit_price: 24.99, currency: "USD" },
    "GIZMO-7": { unit_price: 4.99, currency: "USD" },
    "DOOHICKEY-99": { unit_price: 149.95, currency: "USD" },
    "THINGAMAJIG-5": { unit_price: 59.99, currency: "USD" },
  };

  const entry = catalog[sku.toUpperCase()];
  if (entry) {
    return { sku: sku.toUpperCase(), unit_price: entry.unit_price, currency: entry.currency };
  }

  // Default fallback for unknown SKUs
  return { sku: sku.toUpperCase(), unit_price: 0.0, currency: "USD" };
}

File: scratchpad/failed-jobs/2026-06-11__17-26-28/ai_agent_mcp_tool_definition__fLDJzZb/artifacts/user/wm-agent/f/agent_zr_fldjzzb/lookup_sku.py
def main(sku: str):
    # Returns structured data about a product SKU
    sku_upper = sku.strip().upper()
    if "WIDGET-42" in sku_upper:
        return {
            "sku": "WIDGET-42",
            "unit_price": 42.0,
            "currency": "USD"
        }
    elif "WIDGET" in sku_upper:
        return {
            "sku": sku_upper,
            "unit_price": 19.99,
            "currency": "USD"
        }
    else:
        return {
            "sku": sku_upper,
            "unit_price": 9.99,
            "currency": "USD"
        }
